### PR TITLE
Add DENY privilege statement

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by martint** [comment]
Syntax looks good.

**Review by kokosing — COMMENTED** [review]
Can we have tests like `io.trino.execution.TestGrantOnSchema` and `io.trino.execution.TestGrantOnTable`? Or maybe extend existing ones?

File: core/trino-parser/src/test/java/io/trino/sql/parser/TestStatementBuilder.java [hunk 1]
```diff
@@ -286,6 +286,9 @@ public void testStatementBuilder()
         printStatement("grant select on foo to alice with grant option");
         printStatement("grant all privileges on foo to alice");
         printStatement("grant delete, select on foo to role public");
+        printStatement("deny select on foo to alice");
+        printStatement("deny all privileges on foo to alice");
+        printStatement("deny delete, select on foo to role public");
```
Comment (kokosing) [inline core/trino-parser/src/test/java/io/trino/sql/parser/TestStatementBuilder.java:291]: Add cases with schema

File: testing/trino-tests/src/test/java/io/trino/tests/AbstractTestEngineOnlyQueries.java [hunk 2]
```diff
@@ -1468,6 +1468,7 @@ public void testDescribeOutputNonSelect()
         assertDescribeOutputEmpty("COMMIT");
         assertDescribeOutputEmpty("ROLLBACK");
         assertDescribeOutputEmpty("GRANT INSERT ON foo TO bar");
+        assertDescribeOutputEmpty("DENY INSERT ON foo TO bar");
```
Comment (kokosing) [inline testing/trino-tests/src/test/java/io/trino/tests/AbstractTestEngineOnlyQueries.java:1471]: add deny on schema too

**Review by kokosing — APPROVED** [review]

**Review by electrum — APPROVED** [review]

**Pull request merged**